Modules/posixmodule.c

```diff
@@ -7930,7 +7971,7 @@ static PyObject *
 os_getpid_impl(PyObject *module)
 /*[clinic end generated code: output=9ea6fdac01ed2b3c input=5a9a00f0ab68aa00]*/
 {
-#ifdef MS_WINDOWS_NON_DESKTOP
+#if defined(MS_WINDOWS) && !defined(MS_WINDOWS_DESKTOP)
     return PyLong_FromUnsignedLong(GetCurrentProcessId());
 #else
     return PyLong_FromPid(getpid());
```

Comment:
The CRT headers define `_CRT_USE_WINAPI_FAMILY_DESKTOP_APP` if either no API family is set or if the API family is in the desktop or system API partition. The definition of `_getpid()` is contingent on this macro.

```suggestion
#if defined(MS_WINDOWS) && !defined(MS_WINDOWS_DESKTOP) && !defined(MS_WINDOWS_SYSTEM)
    return PyLong_FromUnsignedLong(GetCurrentProcessId());
#else
    return PyLong_FromPid(getpid());
```

Maybe it's clearer the other way:

```suggestion
#if !defined(MS_WINDOWS) || defined(MS_WINDOWS_DESKTOP) || defined(MS_WINDOWS_SYSTEM)
    return PyLong_FromPid(getpid());
#else
    return PyLong_FromUnsignedLong(GetCurrentProcessId());
```